dockers/MathJaxFonts/buildFonts.sh

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env bash
+
+usage() {
+    while [[ $# -gt 1 ]]; do
+        echo "$1" >&2
+        shift
+    done
+    echo "Usage: ${0##*/} [OPTIONS] [SOURCE]"
+    echo ""
+    echo "SOURCE may be"
+    echo "  - a URL for a tarball, or"
+    echo "  - a local tarball file, or"
+    echo "  - a local directory"
+    echo "with a layout compatible to MathJax-dev."
+    echo "It defaults to ${URL}"
+    echo ""
+    echo "OPTIONS:"
+    echo "  -h|--help         display this help"
+    echo "  --image NAME:TAG  use the named docker image [$IMAGE]"
+    exit $1
+}
+
+used_fonts=(
+    KaTeX_AMS-Regular
+    KaTeX_Caligraphic-Bold
+    KaTeX_Caligraphic-Regular
+    KaTeX_Fraktur-Bold
+    KaTeX_Fraktur-Regular
+    KaTeX_Main-Bold
+    KaTeX_Main-Italic
+    KaTeX_Main-Regular
+    KaTeX_Math-BoldItalic
+    KaTeX_Math-Italic
+    KaTeX_Math-Regular
+    KaTeX_SansSerif-Bold
+    KaTeX_SansSerif-Italic
+    KaTeX_SansSerif-Regular
+    KaTeX_Script-Regular
+    KaTeX_Size1-Regular
+    KaTeX_Size2-Regular
+    KaTeX_Size3-Regular
+    KaTeX_Size4-Regular
+    KaTeX_Typewriter-Regular
+)
+
+filetypes=( ttf eot woff woff2 )
+
+set -e
+cd "$(dirname "$0")"
+
+cleanup() {
+    [[ "${CONTAINER}" ]] \
+        && docker stop "${CONTAINER}" >/dev/null \
+        && docker rm "${CONTAINER}" >/dev/null
+    CONTAINER=
+    [[ -f "${TMPFILE}" ]] && rm "${TMPFILE}"
+    TMPFILE=
+}
+CONTAINER=
+trap cleanup EXIT
+
+IMAGE="katex/fonts:DF-$(openssl sha1 Dockerfile | tail -c 9)"
+URL=https://github.com/Khan/MathJax-dev/archive/master.tar.gz
+TMPFILE=
+FILE=
+NARGS=0
+while [[ $# -gt 0 ]]; do
+    case "$1" in
+        -h|--help)
+            usage 0
+            ;;
+        --image=*)
+            IMAGE="${1#*=}"
+            ;;
+        --image)
+            shift
+            IMAGE="$1"
+            ;;
+        -*)
+            usage "Invalid option: $1" "" 1
+            ;;
+        *)
+            case $NARGS in
+                0)
+                    if [[ -d "$1" ]]; then
+                        TMPFILE="$(mktemp "${TMPDIR:-/tmp}/mjf.XXXXXXXX")"
+                        FILE="$TMPFILE"
+                        pushd "$1"
+                        if [[ ! -f fonts/OTF/TeX/Makefile ]]; then
+                            echo "$1 does not look like MathJax-dev" >&2
+                            exit 1
+                        fi
+                        tar cf "$FILE" Makefile default.cfg fonts/OTF/TeX
+                        popd
+                    elif [[ -f "$1" ]]; then
+                        FILE="$1"
+                    elif [[ "$1" = http?(s)://* ]]; then
```

Comment:
I tried running `./buildFonts.sh /local/path/to/MathJax-dev` and got the following error:
```
./buildFonts.sh: line 97: syntax error in conditional expression: unexpected token `('
```